data/txntest/txn.go

```diff
@@ -269,19 +269,20 @@ func SignedTxns(txns ...*Txn) []transactions.SignedTxn {
 	txgroup := transactions.TxGroup{
 		TxGroupHashes: make([]crypto.Digest, len(txns)),
 	}
+	stxns := make([]transactions.SignedTxn, len(txns))
 	for i, txn := range txns {
-		txn.Group = crypto.Digest{}
-		txgroup.TxGroupHashes[i] = crypto.HashObj(txn.Txn())
+		stxns[i] = txn.SignedTxn()
```

Comment:
Why flip the order? I guess I don't understand this change

Reply:
I get the impression that the intent was to get things ready so that once the group was calculated it could be put into the txn objects and the stxn objects, saving a loop.  I think that would work.  But this change doesn't save a loop, because of the loop used to initialize stxns.  I think that loop could be removed, and the initialization done in the final loop (what would now be line 282.5). Then you would indeed have the benefit of having eliminated a loop.
(Though I should point out this is entirely test code - note sure it needs to be faster.)

Reply:
Fewer copies. The local `Txn` object does a bunch of work inside `.Txn()` or `.SignedTxn()` to construct the `data/transactions` object. This ordering eliminates a call to `.Txn()` to build an object, and just uses thee `data/transactions.SignedTxn` to get the txid from.